File: 20230418_Tuesday/20230418.go
// Leetcode: 1768. Merge Strings Alternately.
// https://leetcode.com/problems/merge-strings-alternately/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 2 ms, faster than 60.00% of Go online submissions for Merge Strings
// Alternately.
// Memory Usage: 2.1 MB, less than 45.00% of Go online submissions for Merge
// Strings Alternately.
// 2023.04.18 Daily Challenge.
package main

func mergeAlternately(word1 string, word2 string) string {
	i, j, k := 0, 0, 0
	m, n := len(word1), len(word2)
	ans := make([]byte, m+n)
	for i < m && j < n {
		ans[k] = word1[i]
		k += 1
		i += 1
		ans[k] = word2[j]
		k += 1
		j += 1
	}
	for i < m {
		ans[k] = word1[i]
		k += 1
		i += 1
	}
	for j < n {
		ans[k] = word2[j]
		k += 1
		j += 1
	}
	return string(ans)
}

File: 20230418_Tuesday/20230418.rb
# Leetcode: 1768. Merge Strings Alternately.
# https://leetcode.com/problems/merge-strings-alternately/
# = = = = = = = = = = = = = =
# Accepted.
# Thanks God, Jesus Christ!
# = = = = = = = = = = = = = =
# Runtime: 80 ms, faster than 85.00% of Ruby online submissions for Merge
# Strings Alternately.
# Memory Usage: 211 MB, less than 35.00% of Ruby online submissions for Merge
# Strings Alternately.
# 2023.04.18 Daily Challenge.
# @param {String} word1
# @param {String} word2
# @return {String}
def merge_alternately(word1, word2)
  flag = true
  i,j = 0,0
  ans = []
  while i < word1.size && j < word2.size
    if flag
      ans.push(word1[i])
      i += 1
    else
      ans.push(word2[j])
      j += 1
    end
    flag = !flag
  end
  while i < word1.size
    ans.push(word1[i])
    i += 1
  end
  while j < word2.size
    ans.push(word2[j])
    j += 1
  end
  ans.join
end
